mmv1/products/gkeonprem/VmwareNodePool.yaml

```diff
@@ -0,0 +1,286 @@
+    imports: "name"
+    description: "The cluster this node pool belongs to."
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "displayName"
+    description: The display name for the node pool.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "annotations"
+    description: |
+      Annotations on the node Pool.
+      This field has the same restrictions as Kubernetes annotations.
+      The total size of all keys and values combined is limited to 256k.
+      Key can have 2 segments: prefix (optional) and name (required),
+      separated by a slash (/).
+      Prefix must be a DNS subdomain.
+      Name must be 63 characters or less, begin and end with alphanumerics,
+      with dashes (-), underscores (_), dots (.), and alphanumerics between.
+    default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "nodePoolAutoscaling"
+    description: Node Pool autoscaling config for the node pool.
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: "minReplicas"
+        description: Minimum number of replicas in the NodePool.
+      - !ruby/object:Api::Type::Integer
+        name: "maxReplicas"
+        description: Maximum number of replicas in the NodePool.
+  - !ruby/object:Api::Type::NestedObject
+    name: "config"
+    description: The node configuration of the node pool.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: "cpus"
+        description: The number of CPUs for each node in the node pool.
+        default_value: 4
+      - !ruby/object:Api::Type::Integer
+        name: "memoryMb"
+        description: The megabytes of memory for each node in the node pool.
+        default_value: 8192
+      - !ruby/object:Api::Type::Integer
+        name: "replicas"
+        description: The number of nodes in the node pool.
+        default_value: 1
+      - !ruby/object:Api::Type::String
+        name: "imageType"
+        required: true
+        description: |
+          The OS image to be used for each node in a node pool.
+          Currently `cos`, `ubuntu`, `ubuntu_containerd` and `windows` are supported.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: "image"
+        description: The OS image name in vCenter, only valid when using Windows.
+      - !ruby/object:Api::Type::Integer
+        name: "bootDiskSizeGb"
+        description: VMware disk size to be used during creation.
+      - !ruby/object:Api::Type::Array
+        name: "taints"
+        description: The initial taints assigned to nodes of this node pool.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: "key"
```

Comment:
Is it valid to specify a taint entry without a key or value? For example:

```
taints {
  key = "foo"
  value = "bar"
 }
 
taints {
 value = "baz"
}

taints {
  key = "bux"
}

taints {}
```

Reply:
Yes, just like annotations and labels field, empty taints value is allowed in node pool resources

Reply:
Yes, just like annotations and labels field, empty taints value is allowed in node pool resources

Reply:
When you say empty taints value, I mean are individual taints allowed to have a missing key or value! The configuration above is an array with 4 entries, not 4 separate arrays.

Reply:
If that is allowed, can we add tests for each of those cases?

Reply:
key and value must be provided but effect is optional. Set key and value as required. Thanks for pointing this out!